### PR TITLE
DS-3658 Configure ReindexerThread disable reindex

### PR DESCRIPTION
**Pull request by AndrewZachWood**
In response to: https://jira.duraspace.org/browse/DS-3658

This contribution wraps the automatic invocation of the ReindexerThread with a configurable property. Effectively allowing the user to configure the automatic triggering of the ReindexerThread.

### CODE REVIEW

**Comment by terrywbrady** [comment]
@AndrewZWood , nice solution.  Thanks for implementing this.  For compatibility with existing functionality, should the default be set to true?
For my production repository, I will definitely set this to false.  When testing with small, experimental DSpace instances, it is nice to have this on by default.

File: dspace/config/modules/discovery.cfg [hunk 1]
```diff
@@ -22,3 +22,5 @@ discovery.index.projection=dc.title,dc.contributor.*,dc.date.issued
 # 1) you need to set the DiscoverySearchRequestProcessor in the dspace.cfg 
 # 2) to show facet on Site/Community/etc. you need to add a Site/Community/Collection
 #	 Processors plugin in the dspace.cfg
+
+#discovery.autoReindex = false
```
Comment (tdonohue) [inline dspace/config/modules/discovery.cfg]: Please add a comment to describe this new setting.  We always try to document settings in the config file itself via a comment.

Comment (AndrewZWood) [reply dspace/config/modules/discovery.cfg]: Sorry Tim! Corrected: https://github.com/DSpace/DSpace/pull/2184/commits/e744d7731acf3f008ed678a7ac837df60de7180b

File: dspace/config/modules/discovery.cfg [hunk 2]
```diff
@@ -22,3 +22,7 @@ discovery.index.projection=dc.title,dc.contributor.*,dc.date.issued
 # 1) you need to set the DiscoverySearchRequestProcessor in the dspace.cfg 
 # 2) to show facet on Site/Community/etc. you need to add a Site/Community/Collection
 #	 Processors plugin in the dspace.cfg
+
+#Enable the url validation of the search.server setting above.
+#Defaults to true: validation is enabled
+#discovery.solr.url.validation.enabled = true
```
Comment (tdonohue) [inline dspace/config/modules/discovery.cfg]: @AndrewZWood : It looks like your last commit was an accident.  You seem to have replaced the setting with something unrelated :)

**Review by terrywbrady — APPROVED** [review]
Changes look good by review

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me, and this was discussed/approved in today's DevMtg: https://wiki.duraspace.org/display/DSPACE/DevMtg+2019-03-27

I did add a new commit to this PR to enhance the comments in `discovery.cfg` around this new configuration.  This also triggered an updated Travis build.  Once Travis approves, we should merge this (for 6.4) and cherry-pick this over to `master`

**Pull request merged**